bp_fe/src/include/bp_fe_icache.vh

```diff
@@ -0,0 +1,77 @@
+/**
+ * bp_fe_icache.vh
+ *
+ * This file declares the internal structs of the BlackParrot Front End I-Cache
+ * module. For simplicity and flexibility, these structs are parameterized. Each
+ * struct declares its width separately to prevent pre-processor ordering issues.
+ */
+
+`ifndef BP_FE_ICACHE_VH
+`define BP_FE_ICACHE_VH
+
+`ifndef BSG_DEFINES_V
+`define BSG_DEFINES_V
+`include "bsg_defines.v"
+`endif
+
+import bp_common_pkg::*;
+
+/*
+ * bp_fe_icache_pc_gen_s defines the interface between I-Cache and pc_gen. The
+ * I-Cache sends the data (instruction) stored at the virtual pc address to the
+ * pc_gen. The width of data (instruction) is specified by the bp_instr_width_gp parameter.
+ */
+`define declare_bp_fe_icache_pc_gen_s(eaddr_width_p) \
+  typedef struct packed {                            \
+    logic [bp_instr_width_gp-1:0] instr;             \
+    logic [eaddr_width_p-1:0]     addr;              \
+  } bp_fe_icache_pc_gen_s
+
+`define bp_fe_icache_pc_gen_width(eaddr_width_p) \
+  (bp_instr_width_gp+eaddr_width_p)
+
+/*
+ *
+ * bp_fe_icache_meta_data_s is the struct that specifies the format of the
+ * I-Cache meta-data array. The meta-data array contains the auxiliary
+ * information, i.e., dirty bits and LRU bits. A dirty bit indicates whether the
+ * corresponding line is dirty. The LRU bits are used to track the recency of
+ * access to the ways. The meta-data array format (for a 4-way associative 
+ * cache) is as follows:
+ * | LRU | dirty | dirty | dirty | dirty |
+ *       - way_0 - way_1 - way_2 - way_3 - 
+ * For the I-Cache meta-data array, we do not need dirty bits. Upon a miss, the
+ * LCE transmits the meta-data (including LRU and dirty bits) information of the
+ * tag set to the CCE, and the CCE makes the necessary decisions. 
+*/
+`define declare_bp_fe_icache_meta_data_s(lce_assoc_p) \
```

Comment:
The comment implies that some LRU information (i.e., the LRU way ID) will be in this struct, but it does not appear to be defined as a member. Could you clarify the comment and/or struct definition?